src/shr_string_mod.F90

```diff
@@ -1011,6 +1012,106 @@ subroutine shr_string_listGetName(list,k,name,rc)
 
   end subroutine shr_string_listGetName
 
+  !===============================================================================
+  !BOP ===========================================================================
+  !
+  ! !IROUTINE: shr_string_listGetAllNames -- Get names of all fields in list
+  !
+  ! !DESCRIPTION:
+  !     Get names of all fields in list
+  !     \newline
+  !     Allocates the output array
+  !     \newline
+  !     call shr\_string\_listGetAllNames(list,names,rc)
+  !
+  ! !REVISION HISTORY:
+  !     2025-Nov-10 - W. Sacks
+  !
+  ! !INTERFACE: ------------------------------------------------------------------
+
+  subroutine shr_string_listGetAllNames(list,names,rc)
+
+    ! !INPUT/OUTPUT PARAMETERS:
+
+    character(*)                 ,intent(in)  :: list     ! list/string
+    character(*), allocatable    ,intent(out) :: names(:) ! k-th name in list
+    integer(SHR_KIND_IN),optional,intent(out) :: rc       ! return code
+
+    !EOP
+
+    !----- local -----
+    integer(SHR_KIND_IN)   :: num_fields ! number of fields in list
+    integer(SHR_KIND_IN)   :: n          ! current field number
+    integer(SHR_KIND_IN)   :: num_chars  ! number of characters in this field
+    integer(SHR_KIND_IN)   :: i0,i1      ! name = list(i0:i1)
+    integer(SHR_KIND_IN)   :: rCode      ! return code
+    integer(SHR_KIND_IN)   :: t01 = 0    ! timer
+
+    !----- formats -----
+    character(*),parameter :: subName =   "(shr_string_listGetAllNames) "
+    character(*),parameter :: F00     = "('(shr_string_listGetAllNames) ',4a)"
+
+    !-------------------------------------------------------------------------------
+    ! Notes:
+    !-------------------------------------------------------------------------------
+
+    if (debug>1 .and. t01<1) call shr_timer_get(t01,subName)
+    if (debug>1) call shr_timer_start(t01)
+
+    rCode = 0
+
+    !--- check that this is a valid list ---
+    if (.not. shr_string_listIsValid(list,rCode) ) then
+       write(s_logunit,F00) "ERROR: invalid list = ",trim(list)
+       call shr_string_abort(subName//" ERROR: invalid list = "//trim(list))
+    end if
+
+    num_fields = shr_string_listGetNum(list)
+
+    allocate(names(num_fields))
```

Comment:
Not sure what the general standards are for share code, but should we do a `stat` check for the allocate statement here?